src/material/list/list.scss

```diff
@@ -195,3 +196,15 @@ mat-action-list button {
     border: 0;
   }
 }
+
+@include token-utils.use-tokens(m2-mat-list.$prefix,
+  m2-mat-list.get-token-slots()) {
+  .mdc-list-item--with-leading-icon .mdc-list-item__start {
+    @include token-utils.create-token-slot(margin-left, list-item-leading-icon-start-space);
+    @include token-utils.create-token-slot(margin-right, list-item-leading-icon-end-space);
+    [dir='rtl'] & {
```

Comment:
unless you need the added specificity to defeat some other style, consider instead just setting `margin-inline-{start,end}`